### PR TITLE
Fix for create table with with clause

### PR DESCRIPTION
**Pull request by collerek**
Close: #155

### CODE REVIEW

**Comment by collerek** [comment]
I branched it from resolve_subqueries (#154) so after we merge that one this one should have only relevant changes.

File: test/test_create_table.py [hunk 1]
```diff
@@ -65,3 +65,29 @@ def test_create_table_as_select_with_joins():
...
+        "it_id",
+        "*",
+        "table_a.name",
+        "table_b.name",
+        "table_a.age",
+        "table_c.age",
+        "table.it_id",
+        "sub.it_id",
+    ]
+    assert parser.tables == ["xyz", "internal_table", "table_a", "table_b", "table_c"]
```
Comment (macbre) [inline test/test_create_table.py:93]: Can we also assert on query type here?

**Comment by macbre** [comment]
Let's wait for #154 to merged before this one - hence converted this PR to draft.

File: test/test_create_table.py (hunk 1 above)
Comment (macbre) [reply test/test_create_table.py:93]: Actually it'd be a good idea to add this assertions in the test cases above as well 🙂

Comment (collerek) [reply test/test_create_table.py:95]: Sure, done

**Comment by tipanverella** [comment]
As a stargazer :wink: , I was wondering when this pull request was going to be scheduled.

File: .github/workflows/python-ci.yml [hunk 2]
```diff
@@ -68,8 +68,8 @@ jobs:
     - name: Upload coverage report to Coveralls
       if: matrix.python-version == '3.9'
       env:
-        COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
-      run: poetry run coveralls
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: poetry run coveralls --service=github
```
Comment (macbre) [inline .github/workflows/python-ci.yml:72]: Sorry for that and thanks for fixing.

Comment (collerek) [reply .github/workflows/python-ci.yml:72]: No worries 🙂

File: test/test_create_table.py [hunk 3]
```diff
@@ -65,3 +66,40 @@ def test_create_table_as_select_with_joins():
...
+
+
+def test_create_table_as_select_in_parentheses():
+    qry = """
+        CREATE TABLE records AS 
+        (SELECT t.id, t.name, e.name as energy FROM t JOIN e ON t.e_id = e.id)
+        """
+    parser = Parser(qry)
```
Comment (macbre) [inline test/test_create_table.py:103]:
```suggestion
    parser = Parser(qry)
    assert parser.query_type == QueryType.CREATE
```

**Review by macbre — APPROVED** [review]
All good 🙂

File: test/test_create_table.py (hunk 3 above)
Comment (collerek) [reply test/test_create_table.py:103]: Yep it will pass and did just merged

**Comment by macbre** [comment]
Closes #160 too.

**Pull request merged**